geoutils/raster/raster.py

```diff
@@ -3363,96 +3443,186 @@ def split_bands(self: RasterType, copy: bool = False, bands: list[int] | int | N
 
         return raster_bands
 
+    @deprecate(
+        Version("0.3.0"),
+        "Raster.to_points() is deprecated in favor of Raster.to_pointcloud() and " "will be removed in v0.3.",
+    )
+    def to_points(self, **kwargs):  # type: ignore
+
+        self.to_pointcloud(**kwargs)  # type: ignore
+
     @overload
-    def to_points(
+    def to_pointcloud(
         self,
-        subsample: float | int,
+        data_column_name: str = "b1",
+        data_band: int = 1,
+        store_auxiliary_bands: bool = False,
+        subsample: float | int = 1,
+        *,
         as_array: Literal[False] = False,
-        pixel_offset: Literal["center", "corner"] = "center",
+        random_state: np.random.RandomState | int | None = None,
+        force_pixel_offset: Literal["center", "ul", "ur", "ll", "lr"] = "ul",
     ) -> NDArrayNum:
         ...
 
     @overload
-    def to_points(
+    def to_pointcloud(
         self,
-        subsample: float | int,
+        data_column_name: str = "b1",
+        data_band: int = 1,
+        store_auxiliary_bands: bool = False,
+        subsample: float | int = 1,
+        *,
         as_array: Literal[True],
-        pixel_offset: Literal["center", "corner"] = "center",
+        random_state: np.random.RandomState | int | None = None,
+        force_pixel_offset: Literal["center", "ul", "ur", "ll", "lr"] = "ul",
     ) -> Vector:
         ...
 
-    def to_points(
+    @overload
+    def to_pointcloud(
         self,
+        data_column_name: str = "b1",
+        data_band: int = 1,
+        store_auxiliary_bands: bool = False,
         subsample: float | int = 1,
+        *,
         as_array: bool = False,
-        pixel_offset: Literal["center", "corner"] = "center",
+        random_state: np.random.RandomState | int | None = None,
+        force_pixel_offset: Literal["center", "ul", "ur", "ll", "lr"] = "ul",
+    ) -> NDArrayNum | Vector:
+        ...
+
+    def to_pointcloud(
+        self,
+        data_column_name: str = "b1",
+        data_band: int = 1,
+        store_auxiliary_bands: bool = False,
```

Comment:
One question that comes up. For those bands, the name will be "bX" whatever the name chosen for `data_column_name`? Or would it make sense to either use a prefix, or a list of names for multibands?

Reply:
You're right! Just took some more time and extended the functionality to support any `auxiliary_data_band` as a list and `auxiliary_column_names` as a list too. Now the user can control everything! :smile: